### PR TITLE
[Zamba] Support attn_implementation dispatch

### PR DESCRIPTION
**Pull request by YangKai0616**
# What does this PR do?

The Zamba model fails during the initialization phase when using `from_pretrained(..., attn_implementation="sdpa")`. The attention module has been integrated into the transformers attention framework. This PR enables the `_supports_sdpa` and `_supports_flash_attn` flags.

### CODE REVIEW

File: src/transformers/models/zamba/modeling_zamba.py [hunk 1]
```diff
@@ -644,7 +644,7 @@ class ZambaPreTrainedModel(PreTrainedModel):
     _no_split_modules = ["ZambaHybridLayer", "ZambaMambaDecoderLayer"]
     _skip_keys_device_placement = ["past_key_values"]
     _supports_flash_attn = False
```
Comment (zucchini-nlp) [inline src/transformers/models/zamba/modeling_zamba.py]: FA can also be `True`, no?

Comment (YangKai0616) [reply src/transformers/models/zamba/modeling_zamba.py]: > FA can also be `True`, no?

Sure. Modified the flag and fixed a dtype mismatch issue in FA. Local (A100) FA-related UT tests all passed.

Comment (zucchini-nlp) [reply src/transformers/models/zamba/modeling_zamba.py]: hmm why is the device movement needed? FA usually doesn't need anhything like that, can you share which test was that failing

Comment (YangKai0616) [reply src/transformers/models/zamba/modeling_zamba.py]: > hmm why is the device movement needed? FA usually doesn't need anhything like that, can you share which test was that failing

It's a dtype cast, not a device movement. 

The failing test is `tests/models/zamba/test_modeling_zamba.py::ZambaModelTest::test_flash_attn_2_fp32_ln` — Zamba's shared attention has tied weights which are excluded from bnb 4-bit quantization, so `o_proj` stays as a regular `nn.Linear` with fp32 weights (after the test upcasts all params), but flash attention outputs fp16, causing a dtype mismatch. The `.to(hidden_states.dtype)` is a no-op in normal usage and only matters in this edge case.

Comment (vasqu) [reply src/transformers/models/zamba/modeling_zamba.py]: I'd tbh rather skip the test then. It's not a super important test int the first place tbh.

Comment (YangKai0616) [reply src/transformers/models/zamba/modeling_zamba.py]: > I'd tbh rather skip the test then. It's not a super important test int the first place tbh.

Reverted the dtype cast and skipped the test instead.

**Comment by zucchini-nlp** [comment]
@bot \style

**Review by zucchini-nlp — APPROVED** [review]
yep, now lgtm

**Comment by zucchini-nlp** [comment]
@bot /style

**Comment by zucchini-nlp** [comment]
Oke, bot not broken, it was a typo (though the commit made by the bot is weird when PR branch isn't rebased)
